test/store_unittest.cc

```diff
@@ -0,0 +1,288 @@
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::Parser::FromString("{\"f\":\"hello\"}")
+      );
+  MetadataStore::Metadata m2(
+      "default-version2",
+      true,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::Parser::FromString("{\"f\":\"hello\"}")
+      );
+  store->UpdateMetadata(mr1, std::move(m1));
+  store->UpdateMetadata(mr2, std::move(m2));
+  std::map<MonitoredResource, MetadataStore::Metadata> metadata_map =
+      GetMetadataMap();
+  EXPECT_FALSE(metadata_map.empty());
+  EXPECT_EQ("default-version1", metadata_map.at(mr1).version);
+  EXPECT_EQ("default-version2", metadata_map.at(mr2).version);
+  PurgeDeletedEntries();
+  metadata_map = GetMetadataMap();
+  EXPECT_EQ("default-version1", metadata_map.at(mr1).version);
+  EXPECT_THROW(metadata_map.at(mr2), std::out_of_range);
+}
+
+TEST(MetadataTest, ParseMetadata) {
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::Parser::FromString("{\"f\":\"hello\"}")
+      );
+  EXPECT_EQ("default-version", m.version);
+  EXPECT_FALSE(m.is_deleted);
+  EXPECT_FALSE(m.ignore);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+            m.created_at);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+            m.collected_at);
+  const auto& obj = m.metadata->As<json::Object>();
+  EXPECT_TRUE(obj->Has("f"));
+  EXPECT_EQ("hello", obj->Get<json::String>("f"));
+}
+
+TEST(MetadataTest, CloneMetadata) {
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::Parser::FromString("{\"f\":\"hello\"}")
+      );
+  MetadataStore::Metadata m_clone = m.Clone();
+  EXPECT_EQ("default-version", m_clone.version);
+  EXPECT_FALSE(m_clone.is_deleted);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+            m_clone.created_at);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+            m_clone.collected_at);
+  const auto& obj = m_clone.metadata->As<json::Object>();
+  EXPECT_TRUE(obj->Has("f"));
+  EXPECT_EQ("hello", obj->Get<json::String>("f"));
+}
+
+TEST(MetadataTest, TestIgnoredMetadata) {
+  MetadataStore::Metadata m = MetadataStore::Metadata::IGNORED();
+  EXPECT_EQ("", m.version);
+  EXPECT_EQ(time::rfc3339::FromString("0000-00-00T00:00:00.000000000Z"),
+            m.created_at);
+  EXPECT_EQ(time::rfc3339::FromString("0000-00-00T00:00:00.000000000Z"),
+            m.collected_at);
+  EXPECT_EQ("{}", m.metadata->ToString());
+  EXPECT_FALSE(m.is_deleted);
+  EXPECT_TRUE(m.ignore);
```

Comment:
This is the only field worth checking. The rest are implementation details.

Reply:
Done.